bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableConfiguration.java

```diff
@@ -47,42 +49,87 @@
     return null;
   }
 
+  private static Configuration setConnectionClass(Configuration config) {
```

Comment:
I'm not a fan of this approach.  This seems to be a setup for ConnectionFactory, which is understandable.  I'd rather have a getConnectionClass() which returns CONNECTION_CLASS.  ConnectionFactory can use the value, but I don't think we should directly mix ConnectionFactory with BigtableConfiguration.